src/Simplex/Chat/Store/ContactRequest.hs

```diff
@@ -0,0 +1,333 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TupleSections #-}
+{-# OPTIONS_GHC -fno-warn-ambiguous-fields #-}
+
+module Simplex.Chat.Store.ContactRequest
+  ( createOrUpdateContactRequest,
+    setContactRequestAccepted
+  )
+  where
+
+import Control.Monad
+import Control.Monad.Except
+import Control.Monad.IO.Class
+import Crypto.Random (ChaChaDRG)
+import Data.Int (Int64)
+import Data.Time.Clock (getCurrentTime)
+import Simplex.Chat.Protocol (businessChatsVersion, MsgContent)
+import Simplex.Chat.Store.Direct
+import Simplex.Chat.Store.Groups
+import Simplex.Chat.Store.Shared
+import Simplex.Chat.Store.Profiles
+import Simplex.Chat.Types
+import Simplex.Chat.Types.Preferences
+import Simplex.Chat.Types.Shared
+import Simplex.Messaging.Agent.Protocol (InvitationId)
+import Simplex.Messaging.Agent.Store.AgentStore (maybeFirstRow)
+import Simplex.Messaging.Agent.Store.DB (Binary (..), BoolInt (..))
+import qualified Simplex.Messaging.Agent.Store.DB as DB
+import Simplex.Messaging.Crypto.Ratchet (PQSupport)
+import Simplex.Messaging.Version
+import UnliftIO.STM
+#if defined(dbPostgres)
+import Database.PostgreSQL.Simple ((:.) (..))
+import Database.PostgreSQL.Simple.SqlQQ (sql)
+#else
+import Database.SQLite.Simple ((:.) (..))
+import Database.SQLite.Simple.QQ (sql)
+#endif
+
+createOrUpdateContactRequest ::
```

Comment:
moved to new module because it required functions from multiple store modules, which either created cyclic dependencies, or required to move multiple other functions to shared module